syft/frameworks/crypten/context.py

```diff
@@ -156,9 +168,76 @@ def _send_party_info(worker, rank, msg, return_values):
         msg (CryptenInit): message containing the rank, world_size, master_addr and master_port.
         return_values (dict): dictionnary holding return values of workers.
     """
-
+    print(f"Sending info to {rank}")
     response = worker.send_msg(msg, worker)
-    return_values[rank] = response.contents
+    print(f"Got response from {rank}")
+    return_values[rank] = _unpack_values(response.contents, model)
+    print(f"Unpacked from {rank}")
+
+
+def _pack_values(values):
+    """Pack return values to be passed into a queue then sent over the wire.
+    The main goal here is to be able to return torch tensors."""
+
+    packed_values = []
+    # single value
+    if not isinstance(values, tuple):
+        packed_values.append(_pack_value(values))
+    # multiple values
+    else:
+        for value in values:
+            packed_values.append(_pack_value(value))
+    return packed_values
+
+
+def _pack_value(value):
+    if isinstance(value, torch.Tensor):
+        return (PACK_TORCH_TENSOR, value.tolist())
+
+    elif isinstance(value, crypten.nn.Module):
+        if value.encrypted:
+            raise TypeError("Cannot pack an encrypted crypten model.")
+        params = []
+        for p in value.parameters():
+            params.append(p.tolist())
+
+        return (PACK_CRYPTEN_MODEL, params)
+
+    return (PACK_OTHER, value)
+
+
+def _unpack_values(values, model=None):
+    """Unpack return values that are fetched from the queue."""
+
+    unpacked_values = []
+    for value in values:
+        unpacked_values.append(_unpack_value(value, model))
```

Comment:
some small changes also would be required here for more complex nested responses